src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Ports;
+using System.Text;
+using System.Threading;
+using UnitsNet;
+
+namespace Iot.Device.Mhz19b
+{
+    /// <summary>
+    /// MH-Z19B CO2 concentration sensor binding
+    /// </summary>
+    public sealed class Mhz19b : IDisposable
+    {
+        private enum Command : byte
+        {
+            ReadCo2Concentration = 0x86,
+            CalibrateZeroPoint = 0x87,
+            CalibrateSpanPoint = 0x88,
+            AutoCalibrationSwitch = 0x79,
+            DetectionRangeSetting = 0x99
+        }
+
+        private enum MessageFormat
+        {
+            Start = 0x00,
+            SensorNum = 0x01,
+            Command = 0x02,
+            DataHighRequest = 0x03,
+            DataLowRequest = 0x04,
+            DataHighResponse = 0x02,
+            DataLowResponse = 0x03,
+            Checksum = 0x08
+        }
+
+        private const int MessageSize = 9;
+        private bool _shouldDispose = false;
+        private SerialPort _serialPort = null;
+        private Stream _serialPortStream = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Mhz19b"/> class using an existing (serial port) stream.
+        /// </summary>
+        /// <param name="stream">Existing stream</param>
+        /// <param name="shouldDispose">If true, the stream gets disposed when disposing the binding</param>
+        public Mhz19b(Stream stream, bool shouldDispose)
+        {
+            _serialPortStream = stream ?? throw new ArgumentNullException(nameof(stream));
+            _shouldDispose = shouldDispose;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Mhz19b"/> class and creates a new serial port stream.
+        /// </summary>
+        /// <param name="uartDevice">Path to the UART device / serial port, e.g. /dev/serial0</param>
+        /// <exception cref="System.ArgumentException">uartDevice is null or empty</exception>
+        public Mhz19b(string uartDevice)
```

Comment:
I think for consistency you might want to consider taking SerialPort as an input here

Reply:
(optional since I don't have any use case of what potential problems it would solve)

Reply:
I would rather stay with the current constructors. The one you cited hides all the details from the user for convinience. The other one gives all flexibility by accepting an existing stream. 
I don't see any use case for the third variant at the moment.